packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -130,10 +130,12 @@ export function transformMiddleware(
         if (url.startsWith(publicPath)) {
           logger.warn(
             colors.yellow(
-              `files in the public directory are served at the root path.\n` +
-                `Instead of ${colors.cyan(url)}, use ${colors.cyan(
-                  url.replace(publicPath, '/')
-                )}.`
+              'Assets in public cannot be imported from JavaScript.\n' +
+                `Instead of ${colors.cyan(
+                  url
+                )}, put the file in the src directory, and use ${colors.cyan(
+                  url.replace(publicPath, '/src/')
+                )} instead.`
             )
```

Comment:
I have tested this PR with playground/assets.
These ones shows the same error.
https://github.com/vitejs/vite/blob/75c3bf65694bc89b395e03dabb81721871d24a9c/playground/assets/index.html#L259
https://github.com/vitejs/vite/blob/75c3bf65694bc89b395e03dabb81721871d24a9c/playground/assets/index.html#L262
For these ones, I think the original message is better.

So I think it needs `if (isImportRequest(url))`.
Also I think the warning message will be more clear if `removeImportQuery(url)` is used instead of `url`.